### PR TITLE
Issue #839 by Inlead: Role validation on user create.

### PR DESCRIPTION
**Pull request by NicolaeGrigoriu**
http://platform.dandigbib.org/issues/839

### CODE REVIEW

File: modules/ding_user/ding_user.module [hunk 1]
```diff
@@ -937,3 +944,17 @@ function ding_user_form_contact_site_form_alter(&$form, $form_state, $form_id) {
...
+/**
+ * Check if newly created user has assigned new roles.
+ *
+ * @param array $form
+ *   Register form elements.
+ * @param array $form_state
+ *   State of the submitted form.
+ */
+function _ding_user_roles_form_validate(array $form, array &$form_state) {
+  if (!array_filter($form_state['values']['roles_change'])) {
```
Comment (JrgenGNielsen) [inline modules/ding_user/ding_user.module]: check should be on roles, not roles_change - see if at least one more value is selected, apart from " authenticated user"

Comment (NicolaeGrigoriu) [inline modules/ding_user/ding_user.module]: On user registration the roles array is not populated, it is on user edit. For user registration roles_change array is populated. So, only on this array can be verified if a role has been assigned.

Comment (JrgenGNielsen) [inline modules/ding_user/ding_user.module]: quiet right - my bad

**Comment by JrgenGNielsen** [comment]
reviewed and accepted

**Comment by cableman** [comment]
Merged by hand to resolve conflict.

**Pull request merged**